src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,101 +1,108 @@
-import React, { useEffect, useState } from 'react';
-import './PostDetails.scss';
-
-import { Loader } from '../Loader';
+import React, { useState, useEffect } from 'react';
 import { NewCommentForm } from '../NewCommentForm';
-import { getPostDetails } from '../../api/posts';
-import { removePostComment, getPostComments } from '../../api/comments';
+import './PostDetails.scss';
+import { Post, Comment } from '../../react-app-env';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
 
 type Props = {
-  postId: number,
+  postId: number | undefined;
 };
 
-export const PostDetails: React.FC<Props> = ({ postId }) => {
+export const PostDetails: React.FC<Props> = ({
+  postId,
+}) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
+  const [visiblecomments, setVisiblecomments] = useState(false);
   const [postDetails, setPostDetails] = useState<Post>();
-  const [postComments, setpPostComments] = useState<Comment[]>([]);
-  const [isLoading, setIsLoading] = useState(true);
-  const [isHidden, setHidden] = useState(true);
 
-  const fetchComments = async () => {
-    setpPostComments(await getPostComments(postId));
-    setIsLoading(false);
-  };
+  const findPost = async () => {
+    if (postId) {
+      const result = await getPostbyId(postId);
 
-  const fetchPostDetails = async () => {
-    setIsLoading(true);
-    setPostDetails(await getPostDetails(postId));
-    fetchComments();
+      setPostDetails(result);
+    }
   };
 
   useEffect(() => {
-    if (postId !== 0) {
-      fetchPostDetails();
-    }
-  }, [postId]);
+    findPost();
+  }, []);
 
-  return (
-    <div className="PostDetails" data-cy="postDetails">
-      <h2>Post details:</h2>
+  const findcomments = async () => {
+    if (postId) {
+      const result = await getComments(postId);
 
-      {isLoading
-        ? <Loader />
-        : (
-          <>
-            <section className="PostDetails__post">
-              <p>{postDetails?.title}</p>
-            </section>
+      setCommentsList(result);
+    }
+  };
 
-            <section className="PostDetails__comments">
-              {postComments.length === 0
-                ? <p>No comments</p>
-                : (
-                  <>
-                    <button
-                      type="button"
-                      className="button"
-                      onClick={() => setHidden(!isHidden)}
-                    >
-                      {`${isHidden ? 'Hide' : 'Show'} ${postComments.length} comments`}
-                    </button>
+  const deletecomment = async (id: number) => {
+    if (commentsList) {
+      await delComment(id);
+      findcomments();
```

Comment:
There is no need to call findcomments() in the deletecomment() function, as this will cause the server to be re-requested. Instead, you can simply remove the comment from the commentsList.
const deleteComment = async (id: number) => {
  if (commentsList) {
    const updatedCommentsList = commentsList.filter((comment) => comment.id !== id);
    setCommentsList(updatedCommentsList);
    await delComment(id);
  }
};